### PR TITLE
Fix TabControl NREs

### PR DESCRIPTION
**Pull request by hughbe**
## Proposed Changes
- Fix TabControl NREs

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2831)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/TabControl.cs [hunk 1]
```diff
@@ -1228,6 +1214,11 @@ public Rectangle GetTabRect(int index)
 
         protected string GetToolTipText(object item)
         {
+            if (item == null)
+            {
+                throw new ArgumentNullException(nameof(item));
+            }
+
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/TabControl.cs]: Have you considered returning `string.Empty` instead?

Comment (weltkante) [reply src/System.Windows.Forms/src/System/Windows/Forms/TabControl.cs]: If you do change this please keep it consistent, either allow arbitrary items or require them to be valid tab pages. If you stop throwing NRE/ANE on null you should also stop throwing ICE when something else than a TabPage is passed.

Comment (RussKie) [reply src/System.Windows.Forms/src/System/Windows/Forms/TabControl.cs]: Yes, thank you.
I'm seeking clarifications internally on how we deal with cases like this in general.

Comment (RussKie) [reply src/System.Windows.Forms/src/System/Windows/Forms/TabControl.cs]: The general guidance is that it is acceptable to update a `NullReferenceException` to an `ArgumentNullException` (though it is a breaking change). The reasoning is essentially that we believe it’s very unlikely that code is depending on getting an NRE. The customer benefit is that it’s easier to tell what was `null`.

Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/TabControl.cs]: Updated to also validate the type

**Comment by RussKie** [comment]
Docs:
* https://github.com/dotnet/docs/issues/17085
* https://github.com/dotnet/docs/issues/17096

**Pull request merged**